docs/how-to/tuning-guides/mi300x.rst

```diff
@@ -0,0 +1,1428 @@
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="2" python3
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="3" python3
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="4" python3
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="5" python3
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="6" python3
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="7" python3
+   /app/vllm/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+Run two instances of ``llama3-8b`` model at the same time on one single GPU
+by specifying ``--gpu-memory-utilization`` to 0.4 (40%), as below (on GPU
+0):
+
+.. code-block:: shell
+
+   ROCR_VISIBLE_DEVICES=0 python3
+   /app/vllm/benchmarks/benchmark_throughput.py --gpu-memory-utilization
+   0.4 --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES=0 python3
+   /app/vllm/benchmarks/benchmark_throughput.py --gpu-memory-utilization
+   0.4 --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+Similarly, use the ``ROCR_VISIBLE_DEVICES`` environment variable to specify
+which GPU (0-7) will run those instances. For example, the setting
+``ROCR_VISIBLE_DEVICES="4,5,6,7"`` exposes GPUs 4, 5, 6, and 7 to the program.
+Inside vLLM, the GPUs are mapped to ``CUDA_VISIBLE_DEVICES`` as 0, 1, 2, and 3.
+
+Run vLLM on multiple GPUs
+-------------------------
+
+The two main reasons to use multiple GPUs:
+
+*  The model size is too big to run vLLM using one GPU as it results
+   CUDA/HIP Out of Memory.
+
+*  To achieve better latency.
+
+To run one vLLM instance on multiple GPUs, use the ``-tp`` or
+``--tensor-parallel-size`` option to specify multiple GPUs. Optionally, use the
+``ROCR_VISIBLE_DEVICES`` environment variable to specify the GPUs.
+
+For example, we can use two GPUs to start an API server on port 8000 as
+below:
+
+.. code-block:: shell
+
+   python -m vllm.entrypoints.api_server --model /path/to/model --dtype
+   float16 -tp 2 --port 8000 &
+
+If issues occur using multiple GPUs, specify the ``--enforce-eager``
+parameter.
```

Comment:
We can remove this line, as the vllm is updated with latest fix and the flag is no longer needed.

Reply:
Done